pkg/gofr/container/datasources.go

```diff
@@ -22,12 +24,14 @@ type DB interface {
 	Select(ctx context.Context, data interface{}, query string, args ...interface{})
 	HealthCheck() *datasource.Health
 	Dialect() string
+	Close() error
 }
 
 type Redis interface {
 	redis.Cmdable
 	redis.HashCmdable
 	HealthCheck() datasource.Health
+	Close(context.Context) error
 }
 
 type Cassandra interface {
```

Comment:
https://gofr.dev/docs/advanced-guide/injecting-databases-drivers
As those are created, managed and injected by user, releasing resources should be users responsibility
Feel free to document shutdown procedure for those databases here
Using `defer Close() ` directly in main method should work nicely after merging this PR since it will release app.Run() instead of killing/hanging whole process

This is basically reason for this PR - ability to release resources properly inside gofr so user can release manually allocated resources